modules/noise/fastnoise_lite.h

```diff
@@ -0,0 +1,237 @@
+	real_t domain_warp_fractal_lacunarity;
+	real_t domain_warp_fractal_gain;
+
+public:
+	FastNoiseLite();
+	~FastNoiseLite();
+
+	// General noise settings.
+
+	void set_noise_type(NoiseType p_noise_type);
+	NoiseType get_noise_type() const;
+
+	void set_seed(int p_seed);
+	int get_seed() const;
+
+	void set_frequency(real_t p_freq);
+	real_t get_frequency() const;
+
+	void set_in_3d_space(bool p_enable);
+	bool is_in_3d_space() const;
+
+	void set_offset(Vector3 p_offset);
+	Vector3 get_offset() const;
+
+	void set_color_ramp(const Ref<Gradient> &p_gradient);
+	Ref<Gradient> get_color_ramp() const;
+
+	// Fractal specific.
+
+	void set_fractal_type(FractalType p_type);
+	FractalType get_fractal_type() const;
+
+	void set_fractal_octaves(int p_octaves);
+	int get_fractal_octaves() const;
+
+	void set_fractal_lacunarity(real_t p_lacunarity);
+	real_t get_fractal_lacunarity() const;
+
+	void set_fractal_gain(real_t p_gain);
+	real_t get_fractal_gain() const;
+
+	void set_fractal_weighted_strength(real_t p_weighted_strength);
+	real_t get_fractal_weighted_strength() const;
+
+	void set_fractal_ping_pong_strength(real_t p_ping_pong_strength);
+	real_t get_fractal_ping_pong_strength() const;
+
+	// Cellular specific.
+
+	void set_cellular_distance_function(CellularDistanceFunction p_func);
+	CellularDistanceFunction get_cellular_distance_function() const;
+
+	void set_cellular_return_type(CellularReturnType p_ret);
+	CellularReturnType get_cellular_return_type() const;
+
+	void set_cellular_jitter(real_t p_jitter);
+	real_t get_cellular_jitter() const;
+
+	// Domain warp specific.
+
+	void set_domain_warp_enabled(bool p_enabled);
+	bool is_domain_warp_enabled() const;
+
+	void set_domain_warp_type(DomainWarpType p_domain_warp_type);
+	DomainWarpType get_domain_warp_type() const;
+
+	void set_domain_warp_amplitude(real_t p_amplitude);
+	real_t get_domain_warp_amplitude() const;
+
+	void set_domain_warp_frequency(real_t p_frequency);
+	real_t get_domain_warp_frequency() const;
+
+	void set_domain_warp_fractal_type(DomainWarpFractalType p_domain_warp_fractal_type);
+	DomainWarpFractalType get_domain_warp_fractal_type() const;
+
+	void set_domain_warp_fractal_octaves(int p_octaves);
+	int get_domain_warp_fractal_octaves() const;
+
+	void set_domain_warp_fractal_lacunarity(real_t p_lacunarity);
+	real_t get_domain_warp_fractal_lacunarity() const;
+
+	void set_domain_warp_fractal_gain(real_t p_gain);
+	real_t get_domain_warp_fractal_gain() const;
+
+	// Interface methods.
+
+	Ref<Image> get_image(int p_width, int p_height, bool p_invert = false) override;
+	Ref<Image> get_seamless_image(int p_width, int p_height, bool p_invert = false, real_t p_blend_skirt = 0.1) override;
+
+	real_t get_noise_1d(real_t p_x) override;
+
+	real_t get_noise_2dv(Vector2 p_v) override;
+	real_t get_noise_2d(real_t p_x, real_t p_y) override;
+
+	real_t get_noise_3dv(Vector3 p_v) override;
+	real_t get_noise_3d(real_t p_x, real_t p_y, real_t p_z) override;
```

Comment:
None of the `get_noise` functions are constant, that looks like an oversight. They can be, see my implementation https://github.com/Zylann/godot_voxel/blob/4c3777ce00473efb426362b83d3c3d5853a2ced0/util/noise/fast_noise_lite.h#L101

Reply:
You're right, for FastNoiseLite they could be `const`. But then the interface methods would need to be `const` too and I am not sure if we should change it (since someone might want to add a noise generator which uses some cache structure internally or something else which requires write access in the get_noise/get_image functions). Or should we prohibit that as it might lead to bad library/API design?

Reply:
It sounds like bad design to me. I care about this in particular because I use one of these in a multithreaded generator, and seeing a non-const call to a noise class from multiple threads is a bit worrying^^"